lightning/src/util/config.rs

```diff
@@ -126,6 +126,26 @@ pub struct ChannelHandshakeConfig {
 	///
 	/// [`KeysInterface::get_shutdown_scriptpubkey`]: crate::chain::keysinterface::KeysInterface::get_shutdown_scriptpubkey
 	pub commit_upfront_shutdown_pubkey: bool,
+
+	/// `their_channel_reserve_satoshis` is the minimum amount that the other node is to keep as a direct payment.
+	/// This ensures that if our counterparty broadcasts a revoked state, we can punish them by
+	/// claiming at least this value on chain
+	///
+	/// `their_channel_reserve_proportion_millionth` is for us to configure counterparty's channel reserve
+	/// for both outgoing and incoming channels.
+	///
+	/// channel_reserve values greater than 30% could be considered highly unreasonable, since that
+	/// amount can never be used for payments.
+	/// Also, if our selected channel_reserve for counterparty and counterparty's selected channel_reserve
+	/// for us, sum upto or greater than channel_value, channel negotiations will fail.
+	///
+	/// `their_channel_reserve_proportion_millionth` is millionths of a channel_value to use as `their_channel_reserve_satoshis`.
+	///
+	/// Default Value is 1% of channel_value, configured as 10,000 millionth
+	/// Lower Bound on `their_channel_reserve_satoshis` is 1000 sats provided as safe implementation limit.
+	/// Max `their_channel_reserve_satoshis` can be equal to channel_value i.e 1000,000 millionth
+	/// but negotiations will fail in that case.
+	pub their_channel_reserve_proportion_millionth: u32,
```

Comment:
A general question:
Do we really need the extreme precision that representing this value as a millionth gives us?
I think it's quite likely that users will find it quite confusing to set this value, as setting values as millionths is not commonly used.

Would instead using something like basis points provide us with enough precision or not? I think that using basis points is likely less confusing from the users perspective as it's more commonly used.

Reply:
While basis points might provide right/enough precision for this use case, i think we should have a standard way of representing proportions. Millionths was being used at some of other places and i didn't see basis points being used in code. 
I agree on the less confusing part.

Reply:
Good point :)!